src/backend/joanie/core/flows/order.py

```diff
@@ -227,44 +228,24 @@ def failed_payment(self):
         Mark order instance as "failed_payment".
         """
 
-    # ruff: noqa: PLR0911
-    # pylint: disable=too-many-return-statements
     def update(self):
         """
         Update the order state.
         """
-        if self._can_be_state_completed():
-            self.complete()
-            return
-
-        if self.instance.state in [
-            enums.ORDER_STATE_ASSIGNED,
-            enums.ORDER_STATE_TO_SIGN,
-            enums.ORDER_STATE_TO_SAVE_PAYMENT_METHOD,
+        for transition in [
+            self.complete,
+            self.to_sign,
+            self.to_save_payment_method,
+            self.pending,
+            self.pending_payment,
+            self.no_payment,
+            self.failed_payment,
         ]:
```

Comment:
I guess the order is important here. Maybe add a comment to warn about future modification ? How to ensure it is the good order ?

Reply:
The tests should, in theory at least :sweat_smile: